test/test_kube_utils.py

```diff
@@ -0,0 +1,96 @@
+    """Arboretum Kubernetes common utility module tests."""
+
+    def test_get_cluster_resources_success_core_api(self):
+        """Ensure that core API resources can be retrieved."""
+        session = MagicMock()
+        resp = MagicMock()
+        data = {'foo', 'bar'}
+        resp.json = MagicMock(return_value={'items': data})
+        resp.raise_for_status = MagicMock()
+        session.get = MagicMock(return_value=resp)
+        resource_type = 'r1'
+        verify = True
+        resources = get_cluster_resources(
+            session, '', [resource_type], verify=verify
+        )
+        session.get.assert_called_once_with(
+            f'api/v1/{resource_type}', verify=verify
+        )
+        resp.raise_for_status.assert_called_once()
+        self.assertEqual(resources, {resource_type: data})
+
+    def test_get_cluster_resources_success_custom_resource(self):
+        """Ensure that custom resources can be retrieved."""
+        session = MagicMock()
+        resp = MagicMock()
+        data = {'foo', 'bar'}
+        resp.json = MagicMock(return_value={'items': data})
+        resp.raise_for_status = MagicMock()
+        session.get = MagicMock(return_value=resp)
+        resource_type = 'example.com/v1alpha1/mycustomtype'
+        verify = True
+        resources = get_cluster_resources(
+            session, '', [resource_type], verify=verify
+        )
+        session.get.assert_called_once_with(
+            f'apis/{resource_type}', verify=verify
+        )
+        resp.raise_for_status.assert_called_once()
+        self.assertEqual(resources, {resource_type: data})
+
+    def test_get_cluster_resources_not_found(self):
+        """Ensure that function works for undefined resources."""
+        session = MagicMock()
+        resp = MagicMock()
+        data = {'foo', 'bar'}
+        resp.json = MagicMock(return_value={'items': data})
+        resp.raise_for_status = MagicMock(side_effect=HTTPError())
+        resp.status_code = 404
+        session.get = MagicMock(return_value=resp)
+        resource_types = ['r1']
+        verify = True
+        resources = get_cluster_resources(
+            session, '', resource_types, verify=verify
+        )
+        session.get.assert_called_once_with('api/v1/r1', verify=verify)
+        resp.raise_for_status.assert_called_once()
+        self.assertEqual(resources, {})
+
+    def test_get_cluster_resources_server_error(self):
+        """Ensure that function raises exception for 500 resources."""
+        session = MagicMock()
+        resp = MagicMock()
+        data = {'foo', 'bar'}
+        resp.json = MagicMock(return_value={'items': data})
+        resp.raise_for_status = MagicMock(side_effect=HTTPError())
+        resp.status_code = 500
+        session.get = MagicMock(return_value=resp)
+        resource_types = ['r1']
+        verify = True
+        with self.assertRaises(HTTPError):
+            get_cluster_resources(session, '', resource_types, verify=verify)
```

Comment:
Tests look good and appear to be testing all execution paths.  Just a suggestion: it might be a better to move the following to a `setUp(self):`.  Your tests will be a bit more concise and easier to read.

```python
def setUp(self):
    """Initialize test objects."""
    self.session = MagicMock()
    self.resp = MagicMock()
    self.data = {'foo', 'bar'}
    self.resp.json = MagicMock(return_value={'items': data})
```